waku/v2/protocol/filter/server.go

```diff
@@ -302,12 +308,17 @@ func (wf *WakuFilterFullNode) pushMessage(ctx context.Context, peerID peer.ID, e
 		}
 		logger.Error("pushing messages to peer", zap.Error(err))
 		wf.subscriptions.FlagAsFailure(peerID)
+		if err := stream.Reset(); err != nil {
+			wf.log.Error("resetting connection", zap.Error(err))
+		}
 		return nil
 	}
 
+	stream.Close()
```

Comment:
Not sure. Keeping the stream open for message push would mean keeping a loop on which you'd attempt to read data from the stream on the client side. Seems to me that this is something that should be specified in the RFC. What do you think if we open an issue for discussing this with @jm-clius.

The behavior proposed on this PR is similar to how it was before, except for properly resetting the stream on errors, so depending on how discussion goes, we can create a separate PR with only changes related to reusing streams

Reply:
Agree, We can take this as a separate issue/PR for discussion. 
But i don't think this detail should be part of an rfc. It is more of a implementation detail so that we don't keep opening and closing streams often atleast in scenarios where we know that it would help keeping open.

Reply:
Tracking this in https://github.com/waku-org/go-waku/issues/831